msal-client-credentials-flow/app.js

```diff
@@ -0,0 +1,68 @@
+/*
+ This application demonstrates how to issue a call to a protected API
+ using the client credentials flow.  A request will be issued to
+ Microsoft Graph using the application's own identity.
+*/
+
+// Microsoft Authentication Library (MSAL) for Node.js
+const msal = require('@azure/msal-node')
+
+// Node.js Express Framework
+const express = require('express')
+
+// Used to make the HTTP GET request to the Graph API
+const https = require('https')
+
+// MSAL configs
+const msalConfig = {
+  auth: {
+    // 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+    clientId: '',
+
+    // Client secret 'Value' (not the ID) from 'Client secrets' in app registration in Azure portal
+    clientSecret: '',
+
+    // Full directory URL, in the form of https://login.microsoftonline.com/<tenant>
+    authority: '',
+
+    // 'Object ID' of app registration in Azure portal - this value is a GUID
+    clientObjectId: ''
+  }
+}
+
+// Initialize MSAL
+const msalConfidentialClientApp = new msal.ConfidentialClientApplication(msalConfig)
+
+// In a client credentials flow, the scope is always in the format '<resource>/.default'
+const tokenRequest = {
+  scopes: ['https://graph.microsoft.com/.default']
+}
+
+// Initialize Express
+const app = express()
+
+app.get('/', (req, res) => {
```

Comment:
I'd prefer that we use named routes that indicate their purpose, both in web apps and APIs. Even when there's a single route and that route is anonymous, we can still indicate the endpoint's purpose at a glance to help lower cognitive load.

Maybe something like this (riff on Graph's `/applications`):

`/application/{id}`